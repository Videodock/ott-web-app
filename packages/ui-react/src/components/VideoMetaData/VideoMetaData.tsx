import React from 'react';

import styles from './VideoMetaData.module.scss';

type Props = {
  attributes: string[];
  separator?: string;
};

const VideoMetaData: React.FC<Props> = ({ attributes, separator = 'â€¢' }: Props) => {
  return (
    <div>
      {attributes.map((value, index) => (
        <React.Fragment key={value}>
          <span>{value}</span>
          {index < attributes.length - 1 && (
            <span className={styles.separator} aria-hidden="true">
              {separator}
            </span>
          )}
        </React.Fragment>
      ))}
    </div>
  );
};

export default VideoMetaData;
