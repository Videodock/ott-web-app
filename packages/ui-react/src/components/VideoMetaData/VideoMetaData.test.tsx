import React from 'react';
import { render } from '@testing-library/react';

import VideoMetaData from './VideoMetaData';

describe('<VideoMetaData>', () => {
  test('renders and matches snapshot', () => {
    const attributes = [2023, '10m', 'Comedy'];
    const { container } = render(<VideoMetaData attributes={attributes} separator="|" />);

    expect(container).toMatchSnapshot();
  });
});
