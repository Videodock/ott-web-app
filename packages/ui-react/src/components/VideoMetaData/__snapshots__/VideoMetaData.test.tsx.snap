// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<VideoMetaData> > renders and matches snapshot 1`] = `
<div>
  <span>
    2023
  </span>
  <span
    aria-hidden="true"
    class="_separator_1267df"
  >
    |
  </span>
  <span>
    10m
  </span>
  <span
    aria-hidden="true"
    class="_separator_1267df"
  >
    |
  </span>
  <span>
    Comedy
  </span>
</div>
`;
